include/ginkgo/core/base/math.hpp

```diff
@@ -200,6 +305,94 @@ GKO_INLINE GKO_ATTRIBUTES constexpr bool is_complex()
 }
 
 
+/**
+ * Allows to check if T is a complex or scalar value during compile time by
+ * accessing the `value` attribute of this struct. If `value` is `true`, T is a
+ * complex/scalar type, if it is `false`, T is not a complex/scalar type.
+ *
+ * @tparam T  type to check
+ */
+template <typename T>
+using is_complex_or_scalar_s = detail::is_complex_or_scalar_impl<T>;
+
+/**
+ * Checks if T is a complex/scalar type.
+ *
+ * @tparam T  type to check
+ *
+ * @return `true` if T is a complex/scalar type, `false` otherwise
+ */
+template <typename T>
+GKO_INLINE GKO_ATTRIBUTES constexpr bool is_complex_or_scalar()
+{
+    return detail::is_complex_or_scalar_impl<T>::value;
+}
+
+
+/**
+ * Removes the complex of complex/scalar type or the template parameter of class
+ * by accessing the `type` attribute of this struct.
+ *
+ * @tparam T  type to remove complex
+ */
+using detail::remove_complex_s;
```

Comment:
Why do we need to export that from the `detail` namespace? I don't see any reason to prefer it over `remove_complex`.

Reply:
I used the similar structure on another stuff to make windows SIFNE work properly before.
You are right, we do not use the structure now, so I will remove it.

Reply:
sorry, we do not use any type converter in SFINAE. only something like `is_complex_or_scalar_s` is used in SFINAE.